paws_collector.js

```diff
@@ -461,6 +439,49 @@ class PawsCollector extends AlAwsCollector {
         });
     };
 
+    batchLogProcess(logs, privCollectorState, nextInvocationTimeout, asyncCallback) {
+        let collector = this;
+        return collector.processLog(logs, collector.pawsFormatLog.bind(collector), null, (err) => {
+           
+            if (err) {
+                // MS api return the logs but if build payload size > 10MB, our buildPayload method return Max payload issue. So we split logs array the in two part and call the processLog again.
+               
+                if ((typeof err === 'string' && err.indexOf('Maximum payload size exceeded') !== -1)) {
+                    let batchLogs = [];
+                    let halfLogs = [];
+                    let loglength = logs.length / 2;
+                    for (const index in logs) {
```

Comment:
it should be easier to use [`array.slice()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice) to split an array into halves.

Reply:
for example you can have a list of indexes to be handled in a `map`, like:
```
const indexArray = [
    {start: 0, stop: half}, 
    {start:half+1, logs.length}
];

indexArray.map(logsPart => {
    collector.batchLogProcess(logs.slice(logsPart.start, logsPart.stop, ....);
});

```

Reply:
and use `indexArray.reduce()` instead of `indexArray.map()`

Reply:
indexArray.reduce() return the single value.  I am checking  foe Async.reduce().